lib/classes/CLI.test.js

```diff
@@ -302,6 +305,72 @@ describe('CLI', () => {
     });
   });
 
+  describe('#logBreakingChanges()', () => {
+    let consoleLogStub;
+
+    beforeEach(() => {
+      cli = new CLI(serverless);
+      consoleLogStub = sinon.stub(cli, 'consoleLog').returns();
+    });
+
+    afterEach(() => {
+      cli.consoleLog.restore();
+      delete process.env.SLS_IGNORE_WARNING;
+    });
+
+    it('should log breaking changes when they are provided', () => {
+      const nextVersion = 'Next';
+
+      cli.breakingChanges = [
+        'x is broken',
+        'y will be updated',
+      ];
+
+      let expectedMessage = '\n';
+      expectedMessage += chalk.yellow(`  WARNING: You are running v${serverlessVersion}. v${nextVersion} will include the following breaking changes:\n`);; //eslint-disable-line
+      expectedMessage += chalk.yellow('  - x is broken\n');
+      expectedMessage += chalk.yellow('  - y will be updated\n');
+      expectedMessage += '\n';
+      expectedMessage += chalk.yellow('  You can opt-out from these warnings by setting the "SLS_IGNORE_WARNING=*" environment variable.\n'); //eslint-disable-line
```

Comment:
@pmuens Really appreciate your dedication to test coverage and high quality code :1st_place_medal: , but I'm still worried about maintaining those kinda tests, especially the "expected messages" above. If we for some reason change an irrelevant character, those tests would fail and we'll need to spend considerable amount of time looking very closely at the message diff, while at the very end it doesn't really matter much. Doing that for every message we log would be hell, especially error messages.

But let's merge this anyway. It doesn't deserve anymore time.

Thanks @pmuens 🙌